### PR TITLE
DAT-4351 Clicking in search suggestions fix

### PR DESCRIPTION
**Pull request by dianafa**
## Links

* https://wikia-inc.atlassian.net/browse/DAT-4351.

### CODE REVIEW

**Comment by SebastianMarzjan** [comment]
LG

File: front/main/app/components/wikia-search.js [hunk 1]
```diff
@@ -103,9 +103,24 @@ export default Component.extend(NoScrollMixin,
...
+			 *
+			 * @returns {void}
+			 */
 			onInputBlur() {
-				this.set('isInputFocused', false);
-				this.setSearchSuggestionItems();
+				Ember.run.later(this, () => {
+					if (!(this.get('isDestroyed') || this.get('isDestroying'))) {
+						this.setSearchSuggestionItems();
+						this.set('inputFocused', false);
```
Comment (idradm) [inline front/main/app/components/wikia-search.js]: Is this necessary inside the later?

File: front/main/app/components/wikia-search.js [hunk 2]
```diff
@@ -103,9 +103,25 @@ export default Component.extend(NoScrollMixin,
...
+			 * and if component still present - clean the suggestions array.
+			 *
+			 * @returns {void}
+			 */
 			onInputBlur() {
-				this.set('isInputFocused', false);
-				this.setSearchSuggestionItems();
+				this.set('inputFocused', false);
+
+				run.later(this, () => {
```
Comment (rogatty) [inline front/main/app/components/wikia-search.js]: Would `Ember.run.next` work here?

Comment (rogatty) [inline front/main/app/components/wikia-search.js]: Would `Ember.run.next` work here?
http://emberjs.com/api/classes/Ember.run.html#method_next

**Comment by bognix** [comment]
@dianafa try what Igor said about `Ember.run.next`.

If it works please use it, if it doesn't this solution LGTM

**Comment by SebastianMarzjan** [comment]
LGTM

**Comment by bognix** [comment]
LGTM

**Comment by dianafa** [comment]
NOT VALID ANYMORE. Close as won't fix.

**Pull request closed without merging**